ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
+    if (query.limit > 1000 && errorPtr) {
```

Comment:
I thought ARC just inserted calls to `release` when necessary. I don't see how that affects this. `release` will call `dealloc` when the reference count for the object reaches zero, and `NSObject` implements `dealloc` as calling `free` or whatever is needed to deallocate the object.